classes/Data.ts

```diff
@@ -370,14 +370,8 @@ export default class Data {
 			for (const profile of saved.profiles) {
 				// If the profile is somehow already added to newData, skip it
 				if (newData.profiles.some(p => p.profile_id === profile.profile_id)) continue;
-
 				// Check if the profile exists on the new data (it should unless deleted)
 				const freshProfile = fresh.profiles.find(p => p.profile_id === profile.profile_id);
-				if (!freshProfile) {
```

Comment:
This statement here does two things.

1. If the profile doesn't exist on the new API response then it still adds it with API off.
2. Exits this iteration of the loop.

This pull request is intended to remove step 1, but step 2 is still critical